crates/uv/src/commands/workspace/run.rs

```diff
@@ -0,0 +1,340 @@
+use std::ffi::OsString;
+use std::path::PathBuf;
+use std::{env, iter};
+
+use anyhow::Result;
+use itertools::Itertools;
+use owo_colors::OwoColorize;
+use tempfile::{tempdir_in, TempDir};
+use tokio::process::Command;
+use tracing::debug;
+
+use distribution_types::{IndexLocations, Resolution};
+use install_wheel_rs::linker::LinkMode;
+use uv_cache::Cache;
+use uv_client::{BaseClientBuilder, RegistryClientBuilder};
+use uv_configuration::{ConfigSettings, NoBinary, NoBuild, PreviewMode, SetupPyStrategy};
+use uv_dispatch::BuildDispatch;
+use uv_fs::Simplified;
+use uv_installer::{SatisfiesResult, SitePackages};
+use uv_interpreter::PythonEnvironment;
+use uv_requirements::{ExtrasSpecification, RequirementsSource, RequirementsSpecification};
+use uv_resolver::{FlatIndex, InMemoryIndex, OptionsBuilder};
+use uv_types::{BuildIsolation, HashStrategy, InFlight};
+use uv_warnings::warn_user;
+
+use crate::commands::{workspace, ExitStatus};
+use crate::printer::Printer;
+
+/// Run a command.
+#[allow(clippy::unnecessary_wraps, clippy::too_many_arguments)]
+pub(crate) async fn run(
```

Comment:
This is just the existing `run.rs`, but moved into the `workspace` module, and the `resolve` and `install` commands were pulled into standalone methods.